atomisticparsers/lammps/parser.py

```diff
@@ -663,44 +648,35 @@ def parse_thermodynamic_data(self):
         if not thermo_data:
             return
 
-        calculation_steps = thermo_data.get('Step')
-        calculation_steps = [int(val) if val is not None else None for val in calculation_steps]
-
-        step_map = {}
-        for i_calc, calculation_step in enumerate(calculation_steps):
-            system_index = self._system_step_map.pop(calculation_steps[i_calc], None)
-            step_map[calculation_step] = {'system_index': system_index, 'calculation_index': i_calc}
-        for step, i_sys in self._system_step_map.items():
-            step_map[step] = {'system_index': i_sys, 'calculation_index': None}
-
-        for step in sorted(step_map):
-            sec_scc = sec_run.m_create(Calculation)
-            sec_scc.step = step
-            sec_scc.time = sec_scc.step * time_step  # TODO Physical times should not be stored for GeometryOpt
-            sec_scc.method_ref = sec_run.method[-1] if sec_run.method else None
-
-            system_index = step_map[step]['system_index']
-            if system_index is not None:
-                sec_scc.forces = Forces(total=ForcesEntry(value=self.traj_parsers.eval('get_forces', system_index)))
-                sec_scc.system_ref = sec_system[system_index]
-
-            calculation_index = step_map[step]['calculation_index']
-            if calculation_index is not None:
-                sec_energy = sec_scc.m_create(Energy)
-                for key, val in thermo_data.items():
-                    key = key.lower()
-                    if key in self._energy_mapping:
-                        sec_energy.contributions.append(
-                            EnergyEntry(kind=self._energy_mapping[key], value=val[calculation_index]))
-                    elif key == 'toteng':
-                        sec_energy.current = EnergyEntry(value=val[calculation_index])
-                        sec_energy.total = EnergyEntry(value=val[calculation_index])
-                    elif key == 'press':
-                        sec_scc.pressure = val[calculation_index]
-                    elif key == 'temp':
-                        sec_scc.temperature = val[calculation_index]
-                    elif key == 'cpu':
-                        sec_scc.time_calculation = float(val[calculation_index])
+        for step in self.thermodynamics_steps:
```

Comment:
ok, I see now that in the super parser the steps function is the union of the lists. But it is still unclear to me that this corresponds to self.thermodynamics_steps, since on line 1330 we have:
`        self.thermodynamics_steps = [int(n) for n in thermo_data.get('Step', [])]`

Please just check this and confirm.

Reply:
the setter function takes in the full thermo steps while the getter funtion returns the sampled steps. same thing for the trajectory. steps correspond to the sampled set of the union of thermo and traj

Reply:
ok, great :+1: